content/blog/2025/04-01_positron_debug/index.md

```diff
@@ -0,0 +1,174 @@
+Once [Jenny](https://bsky.app/profile/jennybryan.bsky.social/post/3lggu2abc3s2y) pointed this out to me, things got _a lot_ easier.
+However, if you've moved the cursor manually, that might not work, for reasons I don't know.
+If I've moved the cursor for some reaason, whenever I do `ctrl` + `enter`, it just is stuck at my `browser()`, without running other code. 
+
+#### History from debugger remains in main environment
+Everything you do while debugging (like running next line code with `n`), [remains in your console history](https://github.com/posit-dev/positron/issues/4478).
+Which means you R history get seriously bloated with non-senscical code.
+For those of us who repeatedly navigate through R history to pick up recent lines of code to rerun, this is quite frustrating.
+
+#### Debugger autofocuses to a temporary function file
+This is a huge annoyance.
+In RStudio, when you enter e debugger in your own functions, you auto-focus on the function in whatever file you have it in.
+In Positron, a [new tab is opened with the contents of the function](https://github.com/posit-dev/positron/issues/3151) as it is, for you to run through, but you cannot alter the code because it's in memory, not an actual file.
+This means that a workflow I use a lot, which is make an empty function, and jump into it to start writing the function while in debugger mode is almost impossible.
+It means jumping back and forth between a 'useless' tab (with the original contents of the function) and where I have defined the function in my `.R`file.
+Hopefully, there will be a better solution for this in the near future.
+
+
+#### Exiting the debugger
+It also tok me a little while to learn how to exit the debugger correctly.
+In RStudio, I have to admit to exiting with the pointy-clicky method in the IDE.
+This is, kind of weird, as I usually always opt for keyboard solutions for things, but I think it was just a thing that has stuck around from my novice days and I never transitioned out of it.
+Now, however, Positron does not necessarily have an super pointy-clicky way to exit, which it great!
+Because now I've learned to exit properly by typing `Q` in the console and hitting enter.
+Much better workflow!
+
+So, while there still are annoying things with the debugger in Positron, I must admit that it has helped me better understand the R debugger as a tool.
+Before, I very much relied on the help RStudio was giving me with debugging, but I think I understand better how it actually works now.
+
+I'd you'd like to see a live demo of the debugger, you can see [James Balamuta's video on YouTube](https://www.youtube.com/watch?v=p_4ZS-nnQ2Q).
+
+### Viewing character delimited files
+
+So, the data viewer is really nice, and mimmicks RStudio's viewer well (with the exception of not showing labels of labelled data).
+It's really nice to explore tabular data in it, though it's not something I often do (I prefer printing to the console).
```

Comment:
btw you can't view lists in the Viewer :crying_cat_face:

Reply:
oh! Yeah, now that you mention it thats annoying! :D I'll add that!